convert/convert.go

```diff
@@ -195,3 +210,29 @@ func removeServiceName(service *file.FService) *file.FService {
 	serviceCopy.ID = kong.String(utils.UUID())
 	return serviceCopy
 }
+
+func convertDistributedToKong3(
+	targetContent *file.Content,
+	outputFilename string,
+	format file.Format,
+) (*file.Content, error) {
+	s, _ := state.NewKongState()
+	rawState, err := file.Get(context.Background(), targetContent, file.RenderConfig{
+		CurrentState: s,
+		KongVersion:  semver.Version{Major: 3, Minor: 0},
+	}, dump.Config{}, nil)
+	if err != nil {
+		return nil, err
+	}
+	targetState, err := state.Get(rawState)
+	if err != nil {
+		return nil, err
+	}
+
+	// file.KongStateToFile calls file.WriteContentToFile
+	return file.KongStateToContent(targetState, file.WriteConfig{
+		Filename:    outputFilename,
+		FileFormat:  format,
```

Comment:
Thinking more about this, I think it would be better to either avoid setting the `_format_version` OR setting it properly based on input from the user (`--to kong-gateway-2.x` or `--to kong-gateway-3.x`).

Imagine having some "distributed" 2.x configuration:

```
services:
- name: svc1
  host: mockbin.org
  routes:
  - name: r1
    paths:
    - /r1
 ```

If we use the new functionality, we'd get something like this:

```
$ ./deck convert --from distributed --to kong-gateway --input-file kong.yaml --yes

$ cat kong.yaml
_format_version: "3.0"
services:
- connect_timeout: 60000
  host: mockbin.org
  name: svc1
  protocol: http
  read_timeout: 60000
  routes:
  - name: r1
    paths:
    - /r1
    preserve_host: false
    protocols:
    - http
    - https
    regex_priority: 0
    strip_path: false
  write_timeout: 60000
```

The input and output files are not the same when deployed to Kong 2.x or 3.x.

I think this may lead to some confusion and undesired outcome for the user.

What do you think?

Reply:
Pushed a change handling this.